syntax = "proto3";
package service;

//service
service Ping {
    //check agent status
  rpc Ping(ReqPingData) returns (RespPingData){}
}

service Deploy {
  rpc Deploy(ReqDeployData) returns (RespDeployData){}
}

service ServiceCheck {
  rpc ServiceCheck(ReqCheckSvcData) returns (RespCheckSvcData){}
}

//args
message ReqPingData {
    string health_url = 1;
}

message RespPingData {
    string status = 1;
}

//
message ReqDeployData {
    string version = 1;
    string type = 2;
    string service_name = 3;
    int32 port = 4;
}

message RespDeployData {
    map<string, string> ret = 1;
}

//
message ReqCheckSvcData {
    string health_url = 1;
}

message RespCheckSvcData {
    string status = 1;
}